src/metpy/calc/thermo.py

```diff
@@ -3264,3 +3264,59 @@ def gradient_richardson_number(height, potential_temperature, u, v, vertical_dim
     dvdz = first_derivative(v, x=height, axis=vertical_dim)
 
     return (mpconsts.g / potential_temperature) * (dthetadz / (dudz ** 2 + dvdz ** 2))
+
+
+@exporter.export
+@preprocess_and_wrap()
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def showalter_index(pressure, temperature, dewpt):
+    """Calculate Showalter Index from pressure temperature and 850 hPa lcl.
+
+    Showalter Index derived from [Galway1956]_:
+    SI = T500 - Tp500
+
+    where:
+    T500 is the measured temperature at 500 hPa
+    Tp500 is the temperature of the lifted parcel at 500 hPa
+
+   Parameters
+   ----------
+
+        pressure : `pint.Quantity`
+            Atmospheric pressure level(s) of interest, in order from highest to
+        lowest pressure
+
+        temperature : `pint.Quantity`
+            Parcel temperature for corresponding pressure
+
+        dewpt (:class: `pint.Quantity`):
+            Parcel dew point temperatures for corresponding pressure
+
+
+    Returns
+    -------
+    `pint.Quantity`
+        Showalter index in delta degrees celsius
+
+    """
+    # find the measured temperature and dew point temperature at 850 hPa.
+    T850, Td850 = interpolate_1d(850 * units.hPa, pressure, temperature, dewpt)
+
+    # find the parcel profile temperature at 500 hPa.
+    Tp500 = interpolate_1d(500 * units.hPa, pressure, temperature)
+
+    # Calculate lcl at the 850 hPa level
+    lcl_calc, _ = lcl(850 * units.hPa, T850[0], Td850[0])
+
+    # Define end height for moist lapse rate calculation
+    p_end = 500 * units.hPa
+
+    # Calculate parcel temp when raised dry adiabatically from surface to lcl
+    dl = dry_lapse(lcl_calc, temperature[0], pressure[0])
+
+    # Calculate parcel temp when raised moist adiabatically from lcl to 500mb
+    ml = moist_lapse(p_end, dl, lcl_calc)
+
+    # Calculate the Showalter index
+    shox = Tp500 - ml
+    return shox
```

Comment:
```suggestion
    return Tp500 - ml
```

Reply:
Do you think the comment for this should be updated? It currently has ``` # Calculate the Showalter index``` but I'm not sure if something should be mentioned about the return aspect, or if the comment should be removed completely? I'm fine leaving it as is, but don't mind updating if necessary!

Reply:
I think the comment is fine, since that is the line calculating the actual index.